coding-tracker/coding-tracker/UserInput.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Globalization;
+
+namespace coding_tracker
+{
+    // Class responsible for receiving user input and validating it
+    static class UserInput
+    {
+        internal static string GetDateInput()
+        {
+            Console.WriteLine("Please insert the date in dd/mm/yyyy form.");
+
+            string? input = Console.ReadLine();
+
+            if (input == "0") return null;
+
+            while (!DateTime.TryParseExact(input, "dd/mm/yyyy", CultureInfo.CurrentCulture, DateTimeStyles.None, out _))
```

Comment:
🔴 **Validation in UserInput.cs**

💡 As per requirements, validation needs to be in it's own class (you will thank me when you get to the UnitTesting project).